pkg/sql/distsqlrun/processors.proto

```diff
@@ -227,6 +227,11 @@ message TableReaderSpec {
   // consumer of this TableReader expects to be able to see in-progress schema
   // changes.
   optional ScanVisibility visibility = 7 [(gogoproto.nullable) = false];
+
+  // If true, instructs the TableReader to not check for misplanned table
+  // readers. Should be set if the TableReader is planned on a node known not
+  // to be the range's leaseholder.
+  optional bool ignore_misplanned_ranges = 8 [(gogoproto.nullable) = false];
```

Comment:
Don't think we should be making this part of the spec. This is only relevant for local `tableReader`s. Not sure if there is another way to pass it down to a processor. Maybe a more general flag on the `flowCtx`?

Reply:
Isn't it also relevant for tableReaders that we know we're not planning correctly, like in the case of the softLimit planning or for unhealthy nodes? I didn't add those to this PR, but I feel like there could be other uses for this. If not, I agree with you and I'll revert.